src/index.js

```diff
@@ -1129,7 +1131,82 @@ const announcement = new Announcement();
             res.redirect(`/say?password=${password}&success=true`);
         });
 
-        // catch all handler to swallow non-crashing rejecions
+        // Serve /envvar form
+        app.get('/envvar', ({ query }, res) => {
+            const { success, password = "" } = /** @type {{ password?:string, success: string }} */(query);
+
+            const validPwd = password === process.env.PASSWORD;
+
+            let kvpHtml = [];
+
+            const statusMap = {
+                true: `<div class="result success">Success! Bot will restart with updated environment variables.</div>`,
+                false: `<div class="result error">Error. Could not perform action.</div>`
+            };
+
+            if(validPwd) {
+                const envVars = getEnvironmentVars();
+
+                // Remove keys that should never be allowed to be displayed/updated via the form
+                const unsafeKeys = [
+                    "ACCOUNT_EMAIL",
+                    "ACCOUNT_PASSWORD",
+                    "NODE_ENV",
+                    "PASSWORD",
+                ];
+                unsafeKeys.forEach(x => delete envVars[x]);
+
+                kvpHtml = Object.keys(envVars).map(key => `<div>${key} <input type="text" name="${key}" value="${envVars[key]}" /></div>`);
+            }
+
+            res.send(`
+                <link rel="icon" href="data:;base64,=" />
+                <link rel="stylesheet" href="css/styles.css" />
+                <h3>Update ElectionBot environment variables</h3>
+                <form method="post">
+                    ${kvpHtml}
+                    <input type="hidden" name="password" value="${password}" />
+                    <button>Submit</button>
+                </form>
+                ${statusMap[success]}
+            `);
+
+            return;
+        });
+        
+        // POST event from /envvar form
+        app.post('/envvar', async ({ body }, res) => {
+            const { password } = /** @type {{ password:string }} */(body);
+
+            const validPwd = password === process.env.PASSWORD;
+
+            // Convert request to JSON object - see https://stackoverflow.com/a/8649003
+            const envVars = JSON.parse('{"' + body.replace(/&/g, '","').replace(/=/g,'":"') + '"}', function(key, value) { return key===""?value:decodeURIComponent(value) });
+
+            // Remove keys that should never be allowed to be displayed/updated via the form
+            const unsafeKeys = [
+                "ACCOUNT_EMAIL",
+                "ACCOUNT_PASSWORD",
+                "NODE_ENV",
+                "PASSWORD",
+            ];
+            unsafeKeys.forEach(x => delete envVars[x]);
```

Comment:
Let's do `every` instead of `forEach` - since `delete` returns `true`/`false`. We *really* do not want to proceed with serving anything if any of the deletions return `false` :) Just in case.